t/t0212-trace2-event.sh

```diff
@@ -323,4 +323,44 @@ test_expect_success 'discard traces when there are too many files' '
 	head -n2 trace_target_dir/git-trace2-discard | tail -n1 | grep \"event\":\"too_many_files\"
 '
 
+# In the following "...redact..." tests, skip testing the GIT_TRACE2_REDACT=0
+# case because we would need to exactly model the full JSON event stream like
+# we did in the basic tests above and I do not think it is worth it.
+
```

Comment:
```
# Furthermore, the Trace2 routines print the same content in normal, perf, or event format, and in t0210 and t0211 we already tested the basic functionality, so no need to repeat it here.
#
# In this test, we use the test-helper to unit test each of the event messages where URLs can appear and confirm that they are redacted in each event.
```

Maybe that goes in the commit message ??

Reply:
sorry to be confusing. my brain was a bit scattered.  let me see what you have below.

Reply:
Let's leave the code comment here as it is.  And then make the commit message something like:

```
t0212: test URL redacting in EVENT format

Test the redact URL changes in the Trace2 EVENT format.

In the previous 2 commits, we tested basic URL redacting functionality
in NORMAL and PERF formats in t0210 and t0211.  Since the Trace2
routines print the same content regardless of format, we do not need
to repeat those tests here.

In this test, we use the test-helper to unit test each of the event
messages where URLs can appear and confirm that they are redacted in
each event.  This allows us to test redaction for events that are difficult
to setup tests for, such as child events.
```

Reply:
no problem. the comment is not really needed with the improved commit message.

Reply:
no problem. the comment is not really needed with the improved commit message.